src/Shared/CoreCLRAssemblyLoader.cs

```diff
@@ -52,7 +58,11 @@ public Assembly LoadFromPath(string fullPath)
             // folders in a NuGet package).
             fullPath = FileUtilities.NormalizePath(fullPath);
 
-            if (Traits.Instance.EscapeHatches.UseSingleLoadContext)
+            // If the requested load comes from the same directory as MSBuild, assume that
+            // the load is part of the platform, and load it using the Default ALC.
+            string assemblyDir = Path.GetDirectoryName(fullPath);
+
+            if (Traits.Instance.EscapeHatches.UseSingleLoadContext || string.Equals(assemblyDir, _msbuildDirPath))
```

Comment:
Is any normalization needed for this path, or can you guarantee that a string comparison is sufficient?  Things that could matter: relative vs full path, casing on some platforms, directory separator normalization, whitespace, extended prefixes (eg `\??\`)

Reply:
I think both should be full paths to the directory with MSBuild, assuming they match, and since they describe the exact same file system path, I think they should be identical in casing, too.

My related question is: is this the right MSBuild.exe? If we're running in, say, VS, this point to the version of MSBuild associated with VS, not the version that's executing at the moment. Is that correct?

Reply:
I have a commit that I'll post shortly that will normalize the path in the same way that the input is normalized, just to be sure.  The cost should be insignificant, as we only do it once per instance of MSBuild.

@forgind on your related question, I believe that this is the right MSBuild based on what you're saying.  We'd want to be pointing at the VS MSBuild in that case, right?  That would ensure that we only load DLLs in the task ALC for tasks that aren't "part of MSBuild".

Reply:
I don't know, but I trust you.

Reply:
Is it easy to test this?  Things like:
```
Task.dll
<msbuildDir>\sub\..\Task.dll
C:\<msBuildPath>\Task.dll
\<msbuildPath>\Task.dll
```
Presumably the path to the task is coming from the project/props/targets so it is user specified.  I think there's also a version of `UsingTask` that uses StrongName.  Does that need to be considered? 

Assuming that can be fed in at the right level.